api/v1alpha3/types.go

```diff
@@ -57,6 +57,10 @@ type NetworkSpec struct {
 	// APIServerLB is the configuration for the control-plane load balancer.
 	// +optional
 	APIServerLB LoadBalancerSpec `json:"apiServerLB,omitempty"`
+
+	// PrivateDNSZoneName define the zone name for the Azure Private DNS.
+	// +optional
+	PrivateDNSZoneName string `json:"privateDNSZoneName,omitempty"`
```

Comment:
We should not change v1alpha3 types in this PR as it's targeting the master branch which implements v1alpha4

Reply:
Also, make sure to add conversion for the new v1alpha4 type in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/api/v1alpha3/azurecluster_conversion.go

You can see examples of restoring a new type at https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/release-0.4/api/v1alpha2/azurecluster_conversion.go#L62 (there are no examples from v1alpha3 to v1alpha4 yet since v1alpha4 types are still new)

Reply:
Just did it, thanks for the link I totally missed this part ^^